property_code_enforcements/make_property_code_enforcements.R

```diff
@@ -26,105 +26,70 @@ raw_data <-
+    c(seq(from = 100, to = nrow(d_address), by = 100), nrow(d_address)),
+    \(x, y) 
+    addr_match_street_name_and_number(
+      x = d_address$addr[x:y], 
+      ref_addr = cagis_addr()$cagis_addr,
+      stringdist_match = "osa_lt_1", 
+      match_street_type = TRUE, 
+      simplify = TRUE
+    ), 
   .progress = TRUE
   )
 
-# collapse back to one tibble
-d <- bind_rows(d_address_list)
+d_address_match_tibble <- purrr::map(d_address_match, \(x) tibble::tibble(cagis_addr = x))
+d_address_match_tibble <- bind_rows(d_address_match_tibble)
 
-d <- 
-  d |>
-  mutate(
-    addr_match_result =
-      case_when(
-        purrr::map_lgl(cagis_addr_matches, is.null) ~ NA,
-        purrr::map_dbl(cagis_addr_matches, vctrs::vec_size) == 0 ~ "no_match",
-        purrr::map_dbl(cagis_addr_matches, vctrs::vec_size) == 1 ~ "single_match",
-        purrr::map_dbl(cagis_addr_matches, vctrs::vec_size) > 1 ~ "multi_match",
-        .default = "foofy"
-      ) |>
-      factor(levels = c("no_match", "single_match", "multi_match"))
-  )
+d_address <- bind_cols(d_address, d_address_match_tibble)
 
-summary(d$addr_match_result) # include in readme (only keep single match)
+saveRDS(d_address, "matched_addr.rds")
 
-raw_data <- 
+d <- 
   raw_data |> 
-  left_join(d, by = "address") 
-
-match_summary <- summary(raw_data$addr_match_result) # include in readme (only keep single match)
-
-glue::glue("There were {prettyNum(nrow(raw_data), ',')} infractions reported between {min(raw_data$ENTERED_DATE)} and {max(raw_data$ENTERED_DATE)}. 
-{prettyNum(match_summary['single_match'], ',')} ({round(match_summary['single_match']/nrow(raw_data)*100)}%) were matched to a single residential address in Hamilton County and were matched to a parcel identifier.
-Note that in the case of condominiums, addresses are matched one-to-one, but are matched to multiple parcel identifiers. 
-The {prettyNum(match_summary['multi_match'], ',')} ({round(match_summary['multi_match']/nrow(raw_data)*100, 1)}%) infractions that were matched to more than one address and 
-{prettyNum(match_summary['no_match'], ',')} ({round(match_summary['no_match']/nrow(raw_data)*100)}%) that were not matched are missing parcel identifier.")
-
-# tummarize enforcements by (CAGIS) address/parcel id
-d_enforcements <- 
-  raw_data |>
-  filter(addr_match_result == "single_match") |>
-  tidyr::unnest(cols = cagis_addr_matches) |>
-  rename(cagis_addr = cagis_addr_matches) |>
-  left_join(cagis_addr, by = "cagis_addr") |> # rep rows for condos.. is this parcel id valid? 
-  tidyr::unnest(cols = cagis_addr_data) |>
-  select(
-    date = ENTERED_DATE,
-    infraction_description = SUB_TYPE_DESC,
-    status = DATA_STATUS_DISPLAY, 
-    address,
-    cagis_addr,
-    cagis_parcel_id
-  ) 
+  left_join(d_address, by = "address") |>
+  left_join(cagis_addr(), by = "cagis_addr") |>
+  mutate(cagis_parcel_id = purrr::map_chr(cagis_addr_data, \(x) ifelse(!is.null(x), x$cagis_parcel_id[1], NA))) |>
```

Comment:
does this mean we select the first cagis parcel id if the addr is matched to more than one?  can we instead choose one randomly, like is done using the `simplify` argument in in `addr::addr_match`?

Reply:
Yes, I meant to go back and change this to randomly choose one. I used the code from `addr_match` to implement.